src/camera.rs

```diff
@@ -331,24 +331,27 @@ impl CameraBuffer<'_> {
         ptr: *mut ffi::vsl_camera_buffer,
         parent: &CameraReader,
     ) -> Result<CameraBuffer, Box<dyn Error>> {
-        // The file descriptor returned by vsl_camera_buffer_dma_fd must be duplicated
-        // so that we can manage ownership within Rust using OwnedFd.
-        let rawfd = unsafe { nix::libc::dup(ffi::vsl_camera_buffer_dma_fd(ptr)) };
-        if rawfd == -1 {
-            let err = io::Error::last_os_error();
-            return Err(Box::new(err));
-        }
-
-        let fd = unsafe { OwnedFd::from_raw_fd(rawfd) };
-        Ok(CameraBuffer { fd, ptr, parent })
+        let original_fd: RawFd = unsafe { ffi::vsl_camera_buffer_dma_fd(ptr) };
+        Ok(CameraBuffer {
+            raw_fd: original_fd,
+            ptr,
+            parent,
+        })
     }
 
     pub fn fd(&self) -> BorrowedFd<'_> {
-        self.fd.as_fd()
+        // the raw_fd allocated by v4l2 will stay valid until the CameraReader is
```

Comment:
This comment should be adjusted to follow the Rust safety comments policy.  Basically just needs to start with `SAFETY:` but otherwise the current commentary is good.

https://std-dev-guide.rust-lang.org/policy/safety-comments.html